notebook/04_portfolio_trade.ipynb

```diff
@@ -4,378 +4,337 @@
-    "\n",
-    "4. Compute the slow and fast exponential moving average and compute the trading signal based on it\n",
-    "\n",
-    "5. Run backtesting and compute the returns from this strategy for each of the days and stock symbols \n",
-    "\n",
+    "4. Compute the slow and fast exponential moving average and compute the trading signal based on it.\n",
+    "5. Run backtesting and compute the returns from this strategy for each of the days and stock symbols.\n",
     "6. Run a simple portfolio optimization by averaging the stocks together for each of the trading days.\n",
+    "7. Compute the sharpe ratio and cumulative return results.\n",
     "\n",
-    "7. Compute the sharpe ratio and cumulative return results\n",
+    "The whole workflow can be organized into a computation graph, which is described in a **yaml** file.\n",
     "\n",
-    "The whole workflow can be organized into a computation graph, which are fully described in a yaml file. Here is snippet of the yaml file:"
+    "Here is snippet of the yaml file:"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 5,
+   "execution_count": 2,
    "metadata": {},
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "- id: node_csvdata\n",
+      "- id:  load_csv_data\n",
       "  type: CsvStockLoader\n",
       "  conf:\n",
       "    path: ./data/stock_price_hist.csv.gz\n",
       "  inputs: []\n",
-      "- id: node_sort\n",
+      "- id: sort\n",
       "  type: SortNode\n",
       "  conf:\n",
       "    keys:\n",
       "      - asset\n",
       "      - datetime\n",
       "  inputs:\n",
-      "    - node_csvdata\n",
-      "- id: node_addReturn\n",
+      "    -  load_csv_data\n",
+      "- id: add_return\n",
       "  type: ReturnFeatureNode\n",
       "  conf: {}\n",
       "  inputs:\n",
-      "    - node_sort\n",
-      "- id: node_addIndicator\n",
-      "  type: AssetIndicatorNode\n",
-      "  conf: {}\n",
-      "  inputs:\n",
-      "    - node_addReturn\n",
-      "- id: node_volumeMean\n",
-      "  type: AverageNode\n",
-      "  conf:\n",
-      "    column: volume\n",
-      "  inputs: \n",
-      "    - node_addIndicator\n"
+      "    - sort\n",
+      "...\n"
      ]
     }
    ],
    "source": [
-    "!head -n 29 ../task_example/port_trade.yaml"
+    "!head -n 18 ../task_example/port_trade.yaml\n",
+    "print(\"...\")"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Each nodes has a unique id, a node type, configuration parameters and input nodes ids. gQuant takes this yaml file, wires it into a graph to visualize it."
+    "Each node is composed of:\n",
+    "- a unique id,\n",
+    "- a node type, \n",
+    "- configuration parameters\n",
+    "- from zero to many input nodes ids.\n",
+    "\n",
+    "gQuant's `load_workflow` and `viz_graph` takes this yaml file, and wires it into a graph.\n",
```

Comment:
"gQuant's `load_workflow` and `viz_graph`" -> " gQuant's `load_taskgraph`"

Reply:
Done.